clang/runtime/dpct-rt/include/rng_utils.hpp.inc

```diff
@@ -239,6 +241,448 @@ private:
+// DPCT_DEPENDENCY_EMPTY
+// DPCT_CODE
+  /// Generate double random number(s) with the distribution 'uniform'.
+  /// \param output The pointer of the first random number.
+  /// \param n The number of random numbers.
+  virtual inline void generate_uniform(double *output, std::int64_t n) = 0;
+// DPCT_LABEL_END
+
+// DPCT_LABEL_BEGIN|rng_generator_base_skip_ahead|dpct::rng::host
+// DPCT_DEPENDENCY_EMPTY
+// DPCT_CODE
+  /// Skip ahead several random number(s).
+  /// \param num_to_skip The number of random numbers skipped.
+  virtual void skip_ahead(const std::uint64_t num_to_skip) = 0;
+// DPCT_LABEL_END
+
+protected:
+  sycl::queue _queue{dpct::get_default_queue()};
+  std::uint64_t _seed{0};
+  std::uint32_t _dimensions{1};
+// DPCT_LABEL_BEGIN|rng_generator_base_1|dpct::rng::host
+// DPCT_DEPENDENCY_BEGIN
+// RngUtils|rng_generator_base
+// DPCT_DEPENDENCY_END
+// DPCT_CODE
+};
+// DPCT_LABEL_END
+
+// DPCT_LABEL_BEGIN|rng_generator_host|dpct::rng::host
+// DPCT_DEPENDENCY_BEGIN
+// RngUtils|rng_generator_host_1
+// DPCT_DEPENDENCY_END
+// DPCT_CODE
+/// The random number generator on host.
+template <typename engine_t = oneapi::mkl::rng::philox4x32x10>
+class rng_generator : public rng_generator_base {
+// DPCT_LABEL_END
+public:
+// DPCT_LABEL_BEGIN|rng_generator_host_constructor|dpct::rng::host
+// DPCT_DEPENDENCY_EMPTY
+// DPCT_CODE
+  /// Constructor of rng_generator.
+  rng_generator()
+      : _engine(std::is_same<engine_t, oneapi::mkl::rng::sobol>::value
+                    ? engine_t(_queue, _dimensions)
+                    : engine_t(_queue, _seed)) {}
+// DPCT_LABEL_END
+
+// DPCT_LABEL_BEGIN|rng_generator_host_set_seed|dpct::rng::host
+// DPCT_DEPENDENCY_EMPTY
+// DPCT_CODE
+  /// Set the seed of host rng_generator.
+  /// \param seed The engine seed.
+  void set_seed(const std::uint64_t seed) {
+    if (seed == _seed) {
+      return;
+    }
+    _seed = seed;
+    rng_generator();
+  }
+// DPCT_LABEL_END
+
+// DPCT_LABEL_BEGIN|rng_generator_host_set_dimensions|dpct::rng::host
+// DPCT_DEPENDENCY_EMPTY
+// DPCT_CODE
+  /// Set the dimensions of host rng_generator.
+  /// \param dimensions The engine dimensions.
+  void set_dimensions(const std::uint32_t dimensions) {
+    if (dimensions == _dimensions) {
+      return;
+    }
+    _dimensions = dimensions;
+    rng_generator();
+  }
+// DPCT_LABEL_END
+
+// DPCT_LABEL_BEGIN|rng_generator_host_set_queue|dpct::rng::host
+// DPCT_DEPENDENCY_EMPTY
+// DPCT_CODE
+  /// Set the queue of host rng_generator.
+  /// \param queue The engine queue.
+  void set_queue(const sycl::queue &queue) {
+    if (queue == _queue) {
+      return;
+    }
+    _queue = queue;
+    rng_generator();
+  }
+// DPCT_LABEL_END
+
+// DPCT_LABEL_BEGIN|rng_generator_host_generate_uniform_bits_uint|dpct::rng::host
+// DPCT_DEPENDENCY_BEGIN
+// RngUtils|rng_generator_host_generate
+// DPCT_DEPENDENCY_END
+// DPCT_CODE
+  /// Generate unsigned int random number(s) with the distribution
+  /// 'uniform_bits'.
```

Comment:
Generate unsigned int random number(s) with the distribution 'uniform_bits' distribution.

Reply:
(All generate methods have this problem.)

Reply:
fixed